File: src/main/java/org/funivan/lologa/algo/ga/genome/value/AverageScoreValue.java
package org.funivan.lologa.algo.ga.genome.value;

import org.funivan.lologa.tile.Score.ScoreSum;
import org.funivan.lologa.tiles.TilesInterface;

public class AverageScoreValue implements ValueInterface {
    @Override
    public String type() {
        return "average_score";
    }

    @Override
    public Double value(TilesInterface original, TilesInterface tiles) {
        double oldAverage = new ScoreSum(original).value() / (double) original.size();
        double newAverage = new ScoreSum(tiles).value() / (double) tiles.size();
        return newAverage - oldAverage;
    }
}

File: src/main/java/org/funivan/lologa/algo/ga/genome/value/AverageScoreValue.kt
package org.funivan.lologa.algo.ga.genome.value

import org.funivan.lologa.tile.Score.ScoreSum
import org.funivan.lologa.tiles.TilesInterface

class AverageScoreValue : ValueInterface {
    override fun type(): String {
        return "average_score"
    }

    override fun value(original: TilesInterface, tiles: TilesInterface): Double? {
        val oldAverage = ScoreSum(original).value() / original.size().toDouble()
        val newAverage = ScoreSum(tiles).value() / tiles.size().toDouble()
        return newAverage - oldAverage
    }
}
